Sources/PackagePlugin/Protocols.swift

```diff
@@ -62,7 +74,35 @@ public protocol CommandPlugin: Plugin {
 }
 
 extension CommandPlugin {
-    
+    /// Compatibility implementation that calls the older version of this method.
+    public func performCommand(
+        /// The context in which the plugin is invoked. This is the same for all
+        /// kinds of plugins, and provides access to the package graph, to cache
+        /// directories, etc.
+        context: PluginContext,
+        
+        /// Any literal arguments passed after the verb in the command invocation.
+        arguments: [String]
+    ) async throws {
+        var targetArgExtractor = ArgumentExtractor(arguments)
+        let targetNames = targetArgExtractor.extractOption(named: "target")
+        let targets: [Target] = try targetNames.map{ name in
+            guard let target = context.package.targets.first(where: { $0.name == name }) else {
+                throw PluginContextError.targetNotFound(packageName: context.package.displayName, targetName: name)
+            }
+            return target
+        }
+        try await self.performCommand(context: context, targets: targets, arguments: targetArgExtractor.remainingArguments)
+    }
+
+    public func performCommand(
```

Comment:
do we need to keep this around? we have not release this api yet, so not sure we need to worry about backwards compatibility

Reply:
I wanted to smooth the transition for DocC, but I think the drawback of having both is a good point.